src/winjs/EngagementProxy.js

```diff
@@ -10,31 +10,33 @@ window.azureEngagement = {
     AZME_ENABLE_NATIVE_LOG: $AZME_ENABLE_NATIVE_LOG,
     connectionString: "$AZME_WINDOWS_CONNECTION_STRING",
     appVersionName: "$APP_VERSION_NAME",
-    appVersionCode: $APP_VERSION_CODE
+	lastActivityName: null,
+	lastActivityUserInfos: null
 };
 
-function engagementLogger(_log)
-{
+function engagementLogger(_log) {
     if (window.azureEngagement.AZME_ENABLE_PLUGIN_LOG)
         console.log("[Engagement Plugin] " + _log);
 }
 
 engagementLogger("AZME pluginVersion v"+window.azureEngagement.pluginVersion);
 
-var script = document.createElement('script');
-script.src =  "plugins/cordova-plugin-ms-azure-mobile-engagement/libs/azure-engagement.js";
-script.onload = function () {
-    engagementLogger("Native Library loaded!");
-}
-document.head.appendChild(script); 
+var xhrObj = new XMLHttpRequest();
+xhrObj.open('GET', "plugins/cordova-plugin-ms-azure-mobile-engagement/libs/azure-engagement.js", false);
+xhrObj.send('');
+var se = document.createElement('script');
+se.type = "text/javascript";
+se.text = xhrObj.responseText;
+document.getElementsByTagName('head')[0].appendChild(se);
+engagementLogger("JS SDK loaded");
 
 cordova.commandProxy.add("Engagement",{ 
 
     startActivity: function (successCallback, errorCallback, _params) {
-        var activityName = _params[0];
-        var userInfos = JSON.parse(_params[1]);
-        engagementLogger("startActivity " + activityName);
-        engagement.agent.startActivity(activityName , userInfos);
+        window.azureEngagement.lastActivityName = _params[0];
+        window.azureEngagement.lastActivityUserInfos = JSON.parse(_params[1]);

```

Comment:
I think the activity name and user info should be cached after being processed by the SDK because the SDK does some validations in it then ignore it and throw an exception in case of errors. That way you'll not restart a faulty activity.
(I would rename userInfos to userInfo as information is a mass noun).